src/gui/painting/qpainter.cpp

```diff
@@ -2673,6 +2673,182 @@ QPainterPath QPainter::clipPath() const
     }
 }
 
+/*!
+    Returns the current clip path in logical coordinates, unrounded.
+
+    The main purpose for this function is inside QSvgGenerator,
+    where we do not want any points to be rounded during the
+    creation of the file.
+
+    \warning QPainter does not store the combined clip explicitly as
+    this is handled by the underlying QPaintEngine, so the path is
+    recreated on demandand transformed to the current logical
+    coordinate system.This is potentially an expensive operation.
+
+    \sa setClipPath(), clipRegion(), setClipping()
+*/
+QPainterPath QPainter::clipPathF() const
+{
+    Q_D(const QPainter);
```

Comment:
I feel like several of the ifs could be removed.

Rather than checking if the iterator is empty or size 1, is it possible to 1. create an empty QPainterPath for the output result, then 2. iterate over the clipInfo vector and update the output state for each entry. If there's only one, there will only be a single iteration, and if it is empty, there will be no iterations and it will output the null path you created.